resteasy-client-reactor-netty/src/main/java/org/jboss/resteasy/client/jaxrs/engines/ReactorNettyClientHttpEngine.java

```diff
@@ -126,8 +152,11 @@ public ReactorNettyClientHttpEngine(final HttpClient httpClient,
                                                         request.getClientConfiguration(),
                                                         response,
                                                         null,
-                                                        extractor)))))
-                .toFuture();
+                                                        extractor)))));
+
+        return timeout
+                .map(timeout -> responseMono.timeout(timeout))
+                .orElse(responseMono).toFuture();
```

Comment:
Please move `toFuture` to the next line.  Along with other comments, I think it reads better as:

```java
return requestTimeout
        .map(duration -> responseMono.timeout(duration))
        .orElse(responseMono)
        .toFuture();
```